subprojects/android/src/main/java/org/mockito/android/internal/creation/AndroidByteBuddyMockMaker.java

```diff
@@ -1,13 +1,32 @@
 package org.mockito.android.internal.creation;
 
 import org.mockito.internal.creation.bytebuddy.SubclassByteBuddyMockMaker;
+import org.mockito.internal.util.ConsoleMockitoLogger;
+import org.mockito.internal.util.Platform;
 import org.mockito.invocation.MockHandler;
 import org.mockito.mock.MockCreationSettings;
 import org.mockito.plugins.MockMaker;
 
-public class AndroidByteBuddyMockMaker implements MockMaker {
+import static org.mockito.internal.util.StringJoiner.join;
 
-    private final MockMaker delegate = new SubclassByteBuddyMockMaker(new AndroidLoadingStrategy());
+public class AndroidByteBuddyMockMaker extends ConsoleMockitoLogger implements MockMaker {
+
+    private final MockMaker delegate;
+
+    public AndroidByteBuddyMockMaker() {
+        if (Platform.isAndroid() || Boolean.getBoolean("org.mockito.mock.android")) {
```

Comment:
@raphw Maybe it is a good idea to move `Boolean.getBoolean("org.mockito.mock.android")` to `Platform.hasDependencyMockitoAndroid()`. Currently it feels a bit cryptic.